### DISCUSSION TITLE
Incompatible PRs

### DISCUSSION THREAD

**Discussion by ykvishal** [idaholab/moose#19337] [Q&A General] [answered]
Hi

I want to use centriod option of FeatureVolumeVectorPostprocessor ( #18672) and KDTree for PolycrystalVoronoi (#17733). However, they are not compatible with each other. I get following errors after adding #17733

> Elem::centroid() has been deprecated. Replace with either Elem::vertex_average() to maintain existing behavior, or the more expensive Elem::true_centroid() in cases where the true 'geometric' centroid is required.

> *** ERROR ***
The following error occurred in the object "grain_volumes", of type "FeatureVolumeVectorPostprocessor".

> While declaring a VectorPostprocessor value with the name "centroid_x",

> a Reporter with the same name has already been declared.

> VectorPostprocessor "grain_volumes/centroid_x":

>   Type:
    std::vector<double>
  Producer:
    FeatureVolumeVectorPostprocessor "grain_volumes"
  Context type:
    ReporterGeneralContext<std::vector<double> >
  Consumer(s):
    None

Is it possible to use these 2 PRs simultaneously ?

Thank you.

Vishal

**Comment by dschwen** [comment, marked as answer]
Yes it is, but you need to replace calls to `centroid()` with calls to `vertex_average()` :-)

**Reply by dschwen** [reply]
To clarify, this was a recent libmesh deprecation. If I remember correctly, `centroid` never calculated the precise centroid of an element anyways, it has always been a vertex average. To avoid further confusion after adding a correct centroid computation method the old method was deprecated and kept under a different name (`vertex_average`), and a new `true_centroid` method was added.
